compiler/noirc_evaluator/src/ssa/opt/inline_functions_with_at_most_one_instruction.rs

```diff
@@ -104,6 +111,69 @@ mod test {
         ");
     }
 
+    #[test]
+    fn does_not_inline_functions_that_require_multiple_passes() {
+        let src = "
+        acir(inline) fn main f0 {
+          b0(v0: Field):
+            v1 = call f2(v0) -> Field
+            return v1
+        }
+
+        acir(inline) fn foo f1 {
+          b0(v0: Field):
+            return v0
+        }
+
+        acir(inline) fn bar f2 {
+          b0(v0: Field):
+            v1 = call f1(v0) -> Field
+            v2 = call f1(v0) -> Field
+            v3 = add v1, v2
+            return v3
+        }
+        ";
+        let ssa = Ssa::from_str(src).unwrap();
+
+        // In the first pass it won't recognize that `main` could be simplified.
+        let ssa = ssa.inline_functions_with_at_most_one_instruction();
+        assert_ssa_snapshot!(ssa.clone(), @r"
```

Comment:
```suggestion
        // In the first pass it won't recognize that `main` could be simplified.
        let ssa = ssa.inline_functions_with_at_most_one_instruction();
        assert_ssa_snapshot!(ssa, @r"
```
I know we have to clone here with the way the test is written right now, but we should be able to avoid cloning the entire SSA.